### DISCUSSION TITLE
Pass parameter from navigation link

### DISCUSSION THREAD

**Discussion by benaunt** [Budibase/budibase#11098] [Help] [answered]
Hi,

Is it possible to pass parameters using the main navigation links, then utilise the values in data provider filters?

I would like to have one screen that adjusts its filtering based on what navigation link is selected. I know I can place buttons or links within the content section of a page, but for this use case the main navigation is how users will want to get to this screen.

The links will be department names like HR, IT etc. Which will open to a summary screen for that department which is the same across each department if filtering can work like this,

Thanks

**Comment by R2bEEaton** [comment, marked as answer]
Hello! Yes, it is possible to use URL parameters to filter data providers, and therefore possible to use navigation links in this way.

### URL Parameters
To do so, first [set up a URL parameter](https://docs.budibase.com/docs/url-parameters) by adding `/:xyz` to the screen you want to have a parameter on. `xyz` can be anything, but the `/:` is required. You can also have multiple parameters. For example, if `employees` is the screen you want to have a table filtered by parameters `a` and `b` on, the entire screen name would be: `/employees/:a/:b`.

Next, use the parameter in the table as one of the filters. `URL.xyz`, where `xyz` is the parameter name, is exposed as a binding you can use with Handlebars or JavaScript. For example using Handlebars, if you had the `:type` parameter, you could use it like this: `{{ URL.type }}`.

![image](https://github.com/Budibase/budibase/assets/34921506/decc623d-c5f8-4356-9d42-c84c43d93d6e)

Then you could make some buttons or links to your screen with the required parameter hardcoded or changed using bindings. Here's an example with hardcoded parameters on buttons with the Navigate To action.

![image](https://github.com/Budibase/budibase/assets/34921506/c4ac29af-2b79-42e6-bb03-f3e7c947a308)

Now when I navigate to the employees page, I see the table filtered by the parameter included in the URL.

Similarly, to answer your specific question: you can use custom links in the navigation bar to achieve the same thing. Clicking on the navigation links now has the same effect as the buttons.

![image](https://github.com/Budibase/budibase/assets/34921506/76c91f7a-14ac-411e-9de8-63834044be7d)

Here is an export of an example app that you can import to explore this in greater detail!

[benaunt-export-1688555814981.tar.gz](https://github.com/Budibase/budibase/files/11957237/benaunt-export-1688555814981.tar.gz)

### Hope that helps!
Let me know if you have any further questions, or mark this question as answered so it can help future users 😃

**Reply by benaunt** [reply]
Thanks so much, managed to implement this into what i was doing perfectly :)
